### PR TITLE
JAMES-2774 Small Cleanup of RabbitMQEventBus

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

**Comment by chibenwa** [comment]
Let's wait a bit to see if reactorrabbit 1.2.0 improves things

**Comment by trantienduchn** [comment]
```
[30b93fca5bc68b4b9845d3e74e7c12be0e5ea57a] Location: /org/apache/james/imap/scripts/ValidateDeployment.test:499
[30b93fca5bc68b4b9845d3e74e7c12be0e5ea57a] LastClientMsg: c SEARCH FLAGGED
[30b93fca5bc68b4b9845d3e74e7c12be0e5ea57a] Expected: '\* SEARCH 1 2 3 4'
[30b93fca5bc68b4b9845d3e74e7c12be0e5ea57a] Actual   : '* SEARCH'
```

**Comment by trantienduchn** [comment]
```
[63467f3ec3818ec4c56b148a95a9334d5ef2bde3] [ERROR] Errors: 
[63467f3ec3818ec4c56b148a95a9334d5ef2bde3] [ERROR]   LinshareBlobExportMechanismProvidingTest.<clinit>:29 » ExceptionInInitializer
```
test this please

**Comment by chibenwa** [comment]
test this please

File: mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/EventDispatcher.java [hunk 1]
```diff
@@ -83,8 +81,8 @@ void start() {
                 dispatchToLocalListeners(event, keys),
                 dispatchToRemoteListeners(serializeEvent(event), keys))
             .subscribeOn(Schedulers.elastic())
+            .doOnError(throwable -> LOGGER.error("error while dispatching event", throwable))
```
Comment (mbaechler) [inline mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/EventDispatcher.java]: How this is related to the comment of this commit?

**Comment by trantienduchn** [comment]
> I see nothing wrong in the code changes but I can't match the comments with the changes. There's nothing redundant in my eyes. Could you explain a bit more?

There is a comment in tests doesn't match with that test behavior now, and there is a unused field in EventDispatcher, I removed them. you can look at the first & second commit. Don't you think they are useless?

**Comment by mbaechler** [comment]
> There is a comment in a test that doesn't match with that test behavior now, and there is a unused field in EventDispatcher, I removed them. you can look at the first & second commit. Don't you think they are useless?

The changes are ok but the comments fail to describe the changes. Please rework the commit comments.

**Comment by trantienduchn** [comment]
```
[ea0a045b8608b44775386d437cfb10f75898b517] [ERROR] Errors: 
[ea0a045b8608b44775386d437cfb10f75898b517] [ERROR]   ManageableMailQueueContract.browseShouldReturnMailsWithAttributes:475 » NoSuchElement
```
test this please

**Comment by trantienduchn** [comment]
```
[ea0a045b8608b44775386d437cfb10f75898b517] [ERROR] Errors: 
[ea0a045b8608b44775386d437cfb10f75898b517] [ERROR]   SpamAssassinContract.deletingSpamMessageShouldNotImpactSpamAssassinState:557 » ConditionTimeout
```
unstable test, but it was green before

**Comment by Arsnael** [comment]
```
SpamAssassinContract.deletingSpamMessageShouldNotImpactSpamAssassinState:557 » ConditionTimeout
```
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**